all_test.go

```diff
@@ -2930,6 +2930,88 @@ func TestMin(t *testing.T) {
 	}
 }
 
+func TestMaxVal(t *testing.T) {
+	tests := []struct{ a, b, c, e int }{
+		{MinInt, MinInt, MinIntM1, MaxInt},
```

Comment:
for each of these test cases, I copied the cases from the corresponding two-arg max/min test and duplicated the first argument of the case (i.e. this case was copied from `TestMax` and went from `{MinInt, MinIntM1, MaxInt},` to `{MinInt, MinInt, MinIntM1, MaxInt},`